qcodes_contrib_drivers/drivers/Attocube/ANC350.py

```diff
@@ -0,0 +1,510 @@
+
+    def _get_frequency(self) -> float:
+        """
+        Returns the frequency parameter of this axis.
+
+        Returns:
+            Frequency in Hertz [Hz], internal resolution is 1 Hz
+        """
+        return self._parent._lib.get_frequency(self._parent._device_handle, self._axis)
+
+    def _set_frequency(self, frequency: float) -> None:
+        """
+        Sets the frequency parameter for this axis
+
+        Args:
+            frequency (float): Frequency in Hertz [Hz], internal resolution is 1 Hz
+        """
+        frequency = int(round(frequency))
+
+        self._parent._lib.set_frequency(self._parent._device_handle, self._axis, frequency)
+
+    def _get_amplitude(self) -> float:
+        """
+        Returns the amplitude parameter of this axis.
+
+        Returns:
+            Amplitude in Volts [V]
+        """
+        return self._parent._lib.get_amplitude(self._parent._device_handle, self._axis)
+
+    def _set_amplitude(self, amplitude: float) -> None:
+        """
+        Sets the amplitude parameter for an axis
+
+        Args:
+            amplitude: Amplitude in Volts [V] (internal resolution is 1mV)
+        """
+        self._parent._lib.set_amplitude(self._parent._device_handle, self._axis, amplitude)
+
+    def _get_status(self) -> Dict[str, bool]:
+        """
+        Reads status information about an axis
+
+        Returns:
+            A Dictionary containing the information about an axis:
+                connected: True, if the axis is connected to a sensor.
+                enabled: True, if the axis voltage output is enabled.
+                moving: True, if the axis is moving.
+                target: True, if the target is reached in automatic positioning.
+                eot_fwd: True, if end of travel detected in forward direction.
+                eot_bwd: True, if end of travel detected in backward direction.
+                error: True, if the axis' sensor is in error state.
+        """
+        keys = ("connected", "enabled", "moving", "target", "eot_fwd", "eot_bwf", "error")
+        status = self._parent._lib.get_axis_status(self._parent._device_handle, self._axis)
+
+        return dict(zip(keys, status))
+
+    def _set_voltage(self, voltage: float) -> None:
+        """
+        Sets the DC level on the voltage output when no sawtooth based motion and no feedback loop
+        is active.
+
+        Args:
+            voltage: DC output voltage in Volts [V], internal resolution is 1 mV
+        """
+        self._parent._lib.set_dc_voltage(self._parent._device_handle, self._axis, voltage)
+
+    def _set_target_position(self, target: float) -> None:
+        """
+        Sets the target position for automatic motion.
+        For linear type actuators the position unit is mm, for goniometers and rotators it is m°.
+
+        Args:
+            target: Target position in millimeters [mm] or millidegrees [m°]. Internal resolution is
+                    1 nm or 1 µ°.
```

Comment:
The internal resolution is nm or µ° if you convert linear and rotational positions with different conversions. Since you have converted with the same factor to mm and m°, shouldn't the resolution to be µm and µ°?

Reply:
The internal resoultion is *internal*. It has nothing to do, with the units returned by this function. The internal resolution indicates how fine the instrument itself can set and measure positions *internally*. That means the instrument can set and measure position down to a precision of 1 nm (linear) or 1 µ° (radiator, goniometer). But those are not the units, the instruments interface uses. The DLL returns/expects meters and degrees. To use values that are a bit easier to handle, it was decided to use mm and °m, but this has nothing to do with the internal resolution.

Reply:
The reason why the internal resolution is written in the documentation string is that the user knows the precision of the instrument. Additionally, he also knows that it brings no advantage to specify values in pm or n° precision, because it exceeds the instruments maximum resolution/precision.

Reply:
I was thinking about the decimal numbers, which raised that question. I am aware of the meaning of resolution and unit on the code side, but thanks for complete explanation. Basically:
we can set for example 2.352367 (in the user side, the unit of this number is mm), which according to the internal precision, it is understandable for the DLL. However, for the rotational position, only first 3 decimals are understandable by the DLL.
My question was about the convenience on the user side, where both linear and rotational position to take the same amount of decimal numbers as shown in the manual.
Sorry if the question was not that clear. But, according to the previous discussions about the conversion factor, there is no need to change anything in this code.